src/scriptdev2/scripts/kalimdor/durotar.cpp

```diff
@@ -130,11 +225,12 @@ bool EffectDummyCreature_lazy_peon_awake(Unit* pCaster, uint32 uiSpellId, SpellE
     // always check spellid and effectindex
     if (uiSpellId == SPELL_AWAKEN_PEON && uiEffIndex == EFFECT_INDEX_0)
     {
+        //TODO some of these checks are now redundant, should be removed by someone who knows how to do so without breaking things
```

Comment:
Let me ask you in another way then, look at line 238. we return false. That's being done when the peon is hit by a wrong spell. the only reason to return true is if the contents of this dummy (line 231 + 232) have been executed, otherwise we should return false.

`pCreatureTarget->GetEntry() != NPC_SLEEPING_PEON` in line 228 is as redundant as can be; the only npc that can run this script is this npc, with that entry, and there is nothing I can think of which would allow that creature to change entry during runtime yet still continue keep it's sd2 script. Even more, if this npc wasn't the target to begin with, this function would have never been executed, no?

Now, for the caster check, since the spell is no longer cast by the npc itself on itself, but only through an item, which is now being checked properly in checkcast, then I don't see it possible that any creature could ever use that spell, the spell is now exclusively related to that item.

and for the aura check, basically it's the same, since it's now being done it checkCast it's just a redundant check since there's no real chance of the mob waking up on it's own in between the time it takes for spell to be cast by the item and when this dummy is triggered.